src/pudl/analysis/clean_up_ferc1.py

```diff
@@ -0,0 +1,655 @@
+# If these columns are nan, we can assume it is either a header row or isn't useful
+nan_cols = [
+    'construction_year',
+    'net_generation_mwh',
+    'total_cost_of_plant',
+    'capex_per_mw',
+    'opex_total',
+    'opex_fuel',
+    'opex_maintenance',
+    'fuel_cost_per_mmbtu'
+]
+
+# If a potential header column has these strings, it's probably a useful header
+header_strings = [
+    'hydro',
+    'hyrdo',
+    'internal',
+    'wind',
+    'solar',
+    'gas',
+    'diesel',
+    'diesal',
+    'steam',
+    'other',
+    'combustion',
+    'combustine',
+    'fuel cell',
+    'hydraulic',
+    'waste',
+    'landfill',
+    'photovoltaic',
+    'nuclear',
+    'oil',
+    'renewable',
+    'facilities',
+    'combined cycle'
+]
+
+# If a potential header has these strings, it is not a header...
+exclude = [
+    '#',
+    r'\*',
+    'pg',
+    'solargenix',
+    'solargennix',
+    r'\@',
+    'rockton',
+    'albany steam'
+]
+
+# ...unless it also has one of these strings
+exceptions = [
+    'hydro plants: licensed proj. no.',
+    'hydro license no.',
+    'hydro: license no.',
+    'hydro plants: licensed proj no.'
+]
+
+# plants with two fuel names in the plant_name
+two_fuel_names_dict = {
+    'las vegas solar': 'solar_pv',
+    'solar centaur': 'gas_turbine'
+}
+
+# What we will rename the headers once we remove them as rows
+new_header_labels = {
+    'hydroelectric': ['hydro', 'hyrdo'],
+    'internal combustion': ['internal', 'interal', 'international combustion'],
+    'combustion turbine': ['combustion turbine'],
+    'combined cycle': ['combined cycle'],
+    'gas turbine': ['gas'],
+    'petroleum liquids': ['oil', 'diesel', 'diesal'],
+    'solar': ['solar', 'photovoltaic'],
+    'wind': ['wind'],
+    'geothermal': ['geothermal'],
+    'waste': ['waste', 'landfill'],
+    'steam': ['steam'],
+    'nuclear': ['nuclear'],
+    'fuel_cell': ['fuel cell'],
+    'other': ['other'],
+    'renewables': ['renewables'],
+}
+
+# Header names that match the one's that zane used in his manual mapping (so we can
+# compare processes)
+zane_header_labels = {
+    'solar_pv': ['solar', 'photovoltaic'],
+    'wind': ['wind'],
+    'hydro': ['hydro', 'hyrdo'],
+    'internal_combustion': ['internal', 'interal', 'international combustion', ],
+    'combustion_turbine': ['combustion turbine', 'combustine turbine'],
+    'combined_cycle': ['combined cycle'],
+    'diesel_turbine': ['oil', 'diesel', 'diesal'],
+    'gas_turbine': ['gas'],
+    'geothermal': ['geothermal'],
+    'waste_heat': ['waste', 'landfill'],
+    'steam_heat': ['steam'],
+    'nuclear': ['nuclear'],
+    'fuel_cell': ['fuel cell'],
+}
+
+
+########################################################################################
+# Helper Functions
+########################################################################################
+
+def expand_dict(dic):
+    """Change format of header_labels."""
```

Comment:
How is this changing the format of `header_labels`?

Reply:
Did my best to explain in the doc string!